### ISSUE TITLE
Idea: stable, opaque identifiers for originator of Flag activities

### ISSUE THREAD

**Issue by jenniferplusplus** [swicg/activitypub-trust-and-safety#8]
It's standard practice for activitypub services to obscure the specific human actor that prompted a Flag activity, and instead use a generic actor. This is a reasonable privacy preserving tactic. It can prevent retaliatory reporting and harassment, for example. However, this also removes a key piece of information that the recipient of a Flag activity would use to evaluate and act on the Flag. It gives the recipient no way to distinguish Flags from multiple actors vs a single spammy actor. It also makes it impossible to correlate Flags across time, which would help to identify victims or participants in harassment campaigns.

As a compromise between anonymity and the needs of community moderators, AP servers could include a stable and reproducible identifier for the Flagging actor, but one which cannot be linked back to them. This could be a URN, to clearly indicate that the value is not resolvable, where the value contains a unique and reproducible key string to represent the originating actor. There are a variety of ways to generate such a key. It could be simply a cryptographically random number. Or a cryptographic hash of the actor's id, as long as it's also salted to prevent recorrelation. The details of how the value is generated are beyond the scope of this proposal, but it's prudent to recommend good practice and signpost potential privacy leaks.

**Comment by nightpool** [comment]
My original proposal when we started federating reports was to only send
reports that were verified/forwarded by your local moderator, with the
reporting instance moderator serving as the point of contact for the report
and removing any anonymity concerns (since the local moderator could simply
be represented by a "system administrator" role account). I think that
would make a lot more sense with the way the network is currently designed
and allow for a lot more communication between instance admins to resolve
moderation issues.

**Comment by ThisIsMissEm** [comment]
See also https://github.com/w3c/activitypub/issues/467#issuecomment-2334599390 and https://socialhub.activitypub.rocks/t/ability-to-distinguish-individual-users-behind-flag-activities-in-a-privacy-preserving-way/4235

**Comment by ThisIsMissEm** [comment]
@nightpool the problem with this:
> only send reports that were verified/forwarded by your local moderator

Is that poorly moderated instances, or instances with low moderator to user counts, would ultimately end up with a backlog of reports not forwarded, and that would actually decrease overall trust & safety of the fediverse, because other servers couldn't pick up the slack to keep their communities safe from harms.

**Comment by nightpool** [comment]
> Is that poorly moderated instances, or instances with low moderator to user counts

Correct. Poorly moderated instances will *always* decrease the overall trust & safety of the fediverse. There is no shortcut to that.
